src/main/java/org/spongepowered/common/data/processor/value/item/BreakableValueProcessor.java

```diff
@@ -0,0 +1,128 @@
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTTagString;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableSetValue;
+import org.spongepowered.api.data.value.mutable.SetValue;
+import org.spongepowered.common.Sponge;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.util.NbtDataUtil;
+import org.spongepowered.common.data.value.mutable.SpongeSetValue;
+
+import java.util.Set;
+
+public class BreakableValueProcessor extends AbstractSpongeValueProcessor<Set<BlockType>, SetValue<BlockType>> {
+
+    public BreakableValueProcessor() {
+        super(Keys.BREAKABLE_BLOCK_TYPES);
+    }
+
+    @Override
+    public SetValue<BlockType> constructValue(Set<BlockType> defaultValue) {
+        return new SpongeSetValue<BlockType>(Keys.BREAKABLE_BLOCK_TYPES, defaultValue);
+    }
+
+    @Override
+    public Optional<Set<BlockType>> getValueFromContainer(ValueContainer<?> container) {
+        if (container instanceof ItemStack) {
+            ItemStack stack = (ItemStack) container;
+            NBTTagCompound tag = stack.getTagCompound();
+            if (tag == null) {
+                return Optional.absent();
+            }
+            NBTTagList blockIds = tag.getTagList("CanDestroy", NbtDataUtil.TAG_STRING);
+            if (blockIds.hasNoTags()) {
+                return Optional.absent();
+            }
+            Set<BlockType> blockTypes = Sets.newHashSetWithExpectedSize(blockIds.tagCount());
+            for (int i = 0; i < blockIds.tagCount(); i++) {
+                Optional<BlockType> blockType = Sponge.getGame().getRegistry()
+                        .getType(BlockType.class, blockIds.getStringTagAt(i));
+                if (blockType.isPresent()) {
+                    blockTypes.add(blockType.get());
+                }
+            }
+            return Optional.of(blockTypes);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(ValueContainer<?> container) {
+        return container instanceof ItemStack;
+    }
+
+    @Override
+    public DataTransactionResult offerToStore(ValueContainer<?> container, Set<BlockType> value) {
+        ImmutableSetValue<BlockType> proposedValue = this.constructValue(value).asImmutable();
+        
+        if (container instanceof ItemStack) {
+            ImmutableSetValue<BlockType> oldValue = getApiValueFromContainer(container).get().asImmutable();
```

Comment:
Since `BreakableData` doesn't exist, you'll need to check if the `getApiValueFromContainer` `Optional` is present.